PyriteEditor/src/editorPasses/PickerPass.h

```diff
@@ -0,0 +1,502 @@
+            {
+                if (!PYR_ENSURE(owner)) return;
+                if (!PYR_ENSURE(boundCamera)) return; // should just get the main camera somehow
+
+                if (pyr::UserInputs::consumeClick(pyr::MouseState::BUTTON_PRIMARY) && ImGui::GetIO().WantCaptureMouse == false)
+                {
+                    computeSelectedActors();
+                }
+
+                if (selectedActors.size() > 0)
+                {
+
+                    RenderSelectedActors();
+
+                    // -- 4. Guizmos
+                    std::vector<Transform*> transformsToEdit;
+                    for (EditorActor* actor : selectedActors)
+                    {
+                        pye::pf_StaticMesh* sm = dynamic_cast<pye::pf_StaticMesh*>(actor);
+                        if (sm)
+                        {
+                            transformsToEdit.push_back(&sm->sourceMesh->getTransform());
+                            continue;
+                        }
+                        pye::pf_BillboardHUD* bb = dynamic_cast<pye::pf_BillboardHUD*>(actor);
+                        if (bb)
+                        {
+                            transformsToEdit.push_back(&((pyr::PointLight*)bb->coreActor)->GetTransform());
+                            continue;
+                        }
+
+                    }
+                    
+                    EditTransforms(*boundCamera, transformsToEdit);
+                }
+            }
+
+        
+        private:
+
+            /// Draws all static meshes and billboards to a 1x1 texture, reads the actor id CPU-side and adds the ID to the selected actor list, pushing on top if CTRL is held.
+            /// This method is quite huge and seems to scale poorly, but i don't really know what else than billboards and static meshes are going to be pickable ?
+            /// 
+            /// TODO : Cycle through actors, not sure how to do this other than excluding the selected actors in the target ?
+            void computeSelectedActors()
+            {
+
+                bool bIsHoldingControl = pyr::UserInputs::isKeyPressed(keys::SC_LEFT_CTRL);
+
+                m_requestedMousePosition = pyr::UserInputs::getMousePosition();
+                pyr::RenderProfiles::pushDepthProfile(pyr::DepthProfile::TESTONLY_DEPTH);
+                m_idTarget.setDepthOverride(m_inputs["depthBuffer"].res.toDepthStencilView());
+
+                auto renderDoc = pyr::RenderDoc::Get();
+                if (renderDoc)    renderDoc->StartFrameCapture(nullptr, nullptr);
+
+                m_idTarget.clearTargets();
+                m_idTarget.bind();
+
+                pcameraBuffer->setData(CameraBuffer::data_t{ .mvp = boundCamera->getViewProjectionMatrix(), .pos = boundCamera->getPosition() });
+                for (const pyr::StaticMesh* smesh : owner->GetContext().ActorsToRender.meshes)
+                {
+                    // Do not render already selected actors ?
```

Comment:
You probably don't want that, clicking on a selected actor should do nothing/remove it from the selection if ctrl is pressed. Not rendering selected actors would make you able to pick through objects, which seems strange

Reply:
Yeah i tried yesterday, and the behaviour was a bit strange. My idea was to "cycle" through objects, not sure how usefull it is tho